### ISSUE TITLE
Reading WAVECAR for SOC is not working

### ISSUE THREAD

**Issue by aamirshafique2977** [materialsproject/pymatgen#1995]
Hello,
I hope you will find this message in good health. 
I trying to load the spin-orbit coupling WAVCAR but failed. I got the following error message: 
Traceback (most recent call last):
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/bin/amset", line 33, in <module>
    sys.exit(load_entry_point('amset', 'console_scripts', 'amset')())
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/amset/amset/tools/wavefunction.py", line 74, in wave
    coeffs, gpoints = _wavefunction_vasp(ibands, planewave_cutoff, **kwargs)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/amset/amset/tools/wavefunction.py", line 93, in _wavefunction_vasp
    wf = get_wavefunction(wavecar=wavecar, directory=directory, vasp_type=vasp_type)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/amset/amset/wavefunction/vasp.py", line 28, in get_wavefunction
    return Wavecar(wavecar, vasp_type=vasp_type)
  File "/lustre/project/k1388/shafiq/CSLD-AMSET/install4/lib/python3.7/site-packages/pymatgen/io/vasp/outputs.py", line 4433, in __init__
    raise ValueError(f'Incorrect value of vasp_type given ({vasp_type}).'
ValueError: Incorrect value of vasp_type given (ncl). Please open an issue if you are certain this WAVECAR was generated with the given vasp_type.

**Comment by mturiansky** [comment]
Can you provide your pymatgen version, VASP version, and a copy of the WAVECAR?

**Comment by aamirshafique2977** [comment]
I have used pymatgen==2020.11.11 and vasp 5.4.4 version. I am unable to attach the WAVECAR file because the size of the file is very large (approx. 155 GB).

**Comment by mturiansky** [comment]
Do you have enough memory? The entire WAVECAR will be read in to memory. I've tested against a WAVECAR generated from 5.4.4, so it should work. Without the WAVECAR, I'm not really sure how to debug it for you.

**Comment by ManojSettipalli** [comment]
Hi Mark, 

I have also run into the 'incorrect vasp_type'

 issue trying to convert WAVECAR to wavefunction.h5 file. The WAVECAR is generated by VASP 5.4.4 and is about 5 MB. So, I do not think mine is a memory issue. Could you test if you could convert my WAVECAR file? I have tested this with pymatgen installed on different OS (both linux and windows). Any insights will be much appreciated.

Thank you,
Manoj

[WAVECAR.gz](https://github.com/materialsproject/pymatgen/files/6651818/WAVECAR.gz)

**Comment by mturiansky** [comment]
Hi @ManojSettipalli, I took a look at the file you sent. It seems that the number of plane waves that are read in from the WAVECAR at the first kpoint is 339, which is not evenly divisible by 2, so something is wrong. A noncollinear WAVECAR has spinors, so there should be twice as many plane waves as usual, one set for spin up and one set for down. By any chance, was this WAVECAR from a relaxation where the volume was also relaxed? That's my only guess as to what may be going wrong. Perhaps try doing a fixed geometry calculation from this WAVECAR to generate a new one and see if that fixes it.

If that's not the case, then I'm not really sure what might be wrong. I can look into it, but I'm quite busy at the moment, so I may not get back to you for some time.

**Comment by ManojSettipalli** [comment]
Hi Mark,

Thank you for your response. I should have mentioned that I got the error for an 'std' type calculation so it is not a noncollinear calculation. I still had this issue. As a last attempt, I fully uninstalled Python on my computer and reinstalled a new python on Conda, and then installed pymatgen. Now, the WAVECAR is being read. So, while I am not entirely sure what the issue was, I am just going with this for now. 

Thank you,
Manoj

**Comment by mturiansky** [comment]
That's strange. I'm glad it worked out in the end.

**Comment by Nokimann** [comment]
I guess this problem may come from the precision of calculating `Gpoints` from `_generate_G_points` function.
As you can find in the code, the error is raised when the deduced `Gpoints` is mismatched with `nplane`.
The plane waves near the boundary of the cutoff radius (in energy) are critical to include as bases.

```
                    if len(self.Gpoints[ink]) != nplane and 2 * len(self.Gpoints[ink]) != nplane:
                        raise ValueError(
                            f"Incorrect value of vasp_type given ({vasp_type})."
                            " Please open an issue if you are certain this WAVECAR"
                            " was generated with the given vasp_type."
                        )
```

So, changing the k-point that encounters the error might solve the problem.

**Comment by janosh** [comment]
@Nokimann Feel free to open a new issue if this is still an issue. I'll close this issue as completed for now.

**Comment by wladerer** [comment]
I have encountered a similar problem on line 4345 of the vasp outputs.py module. I have explicitly included the correct precision and vasp binary type (accurate, ncl). The incar that produced it is the following:

```
ALGO = Fast
AMIN = 0.01
EDIFF = 1e-06
ENCUT = 600 eV
GGA = PE
ISMEAR = 0
LASPH = True
LDAU = False
LMAXMIX = 4
LORBIT = 11
LREAL = Auto
LSORBIT = True
LWAVE = True
MAGMOM = 100*0
NELM = 100
PREC = Accurate
SIGMA = 0.05
```

Do keep in mind that I produced this WAVECAR using line mode, so perhaps that might be an issue? 

For now, I have commented out the conditional statement to see if there are any further issues, as all I need is the charge density at a specific single kpoint.
